backends/s3/s3.go

```diff
@@ -108,37 +108,31 @@ func (b *Backend) List(ctx context.Context, prefix string) (simpleblob.BlobList,
 		return b.doList(ctx, prefix)
 	}
 
+	b.mu.Lock()
```

Comment:
You have moved the lock around network calls that can now block other goroutines for several seconds.